### PR TITLE
131 add updated game carousel

### PR DESCRIPTION
**Pull request by AGI-CEO**
Alright the carousel sale component is nice and animated.
Click the button to go to the greatest repo of all time 🔥

### CODE REVIEW

File: client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
...cmlnaW5hbF9kaXJuYW1lID0gXCIvaG9tZS9ibGFpc2UtbGludXgvY29kZS9TdGVhbS9jbGllbnRcIjtjb25zdCBfX3ZpdGVfaW5qZWN0ZWRfb3JpZ2luYWxfZmlsZW5hbWUgPSBcIi9ob21lL2JsYWlzZS1saW51eC9jb2RlL1N0ZWFtL2NsaWVudC92aXRlLmNvbmZpZy5qc1wiO2NvbnN0IF9fdml0ZV9pbmplY3RlZF9vcmlnaW5hbF9pbXBvcnRfbWV0YV91cmwgPSBcImZpbGU6Ly8vaG9tZS9ibGFpc2UtbGludXgvY29kZS9TdGVhbS9jbGllbnQvdml0ZS5jb25maWcuanNcIjtpbXBvcnQgZG90ZW52IGZyb20gXCJkb3RlbnZcIjtcbmltcG9ydCByZWFjdCBmcm9tIFwiQHZpdGVqcy9wbHVnaW4tcmVhY3Qtc3djXCI7XG5cbmRvdGVudi5jb25maWcoeyBwYXRoOiBcIi4uLy5lbnZcIiB9KTtcblxuZXhwb3J0IGRlZmF1bHQge1xuICBwbHVnaW5zOiBbcmVhY3QoKV0sXG4gIHNlcnZlcjoge1xuICAgIHByb3h5OiB7XG4gICAgICBcIi9hcGlcIjogYGh0dHA6Ly9sb2NhbGhvc3Q6JHtwcm9jZXNzLmVudi5QT1JUfWAsXG4gICAgfSxcbiAgfSxcbiAgY2FjaGVEaXI6IFwiLi4vbm9kZV9tb2R1bGVzLy52aXRlXCIsXG4gIGFzc2V0c0luY2x1ZGU6IFtcIioqLyouZ2xiXCJdLFxufTtcbiJdLAogICJtYXBwaW5ncyI6ICI7QUFBOFIsT0FBTyxZQUFZO0FBQ2pULE9BQU8sV0FBVztBQUVsQixPQUFPLE9BQU8sRUFBRSxNQUFNLFVBQVUsQ0FBQztBQUVqQyxJQUFPLHNCQUFRO0FBQUEsRUFDYixTQUFTLENBQUMsTUFBTSxDQUFDO0FBQUEsRUFDakIsUUFBUTtBQUFBLElBQ04sT0FBTztBQUFBLE1BQ0wsUUFBUSxvQkFBb0IsUUFBUSxJQUFJLElBQUk7QUFBQSxJQUM5QztBQUFBLEVBQ0Y7QUFBQSxFQUNBLFVBQVU7QUFBQSxFQUNWLGVBQWUsQ0FBQyxVQUFVO0FBQzVCOyIsCiAgIm5hbWVzIjogW10KfQo=
```
Comment (gnprwx) [inline client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs:18]: Do we need this mapping url?

File: client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// vite.config.js
+import dotenv from "file:///home/blaise-linux/code/Steam/node_modules/dotenv/lib/main.js";
+import react from "file:///home/blaise-linux/code/Steam/node_modules/@vitejs/plugin-react-swc/index.mjs";
```
Comment (gnprwx) [inline client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs:3]: Don't think these import statements are going to work outside of your computer.

File: client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// vite.config.js
```
Comment (gnprwx) [inline client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs:1]: Do we need this file? I can't see a reason we need it.

**Review by gnprwx — COMMENTED** [review]
Just some file clean up needing to be done and some dependencies already exist in either client or server that have been duplicated in root dependencies. 

Also, keep in mind that these files are too big for production environment. I suggest we keep this code in a separate branch instead of merging into master. That way we can use this branch locally to showcase alongside our master branch which will be hosted online.

**Comment by gnprwx** [comment]
Moving this to a new branch titled 3d-showcase-localhost.

**Comment by gnprwx** [comment]
This is good to merge into the new branch after handling the comments I made about file cleanup and removing redundant dependencies.

**Comment by AGI-CEO** [comment]
> These files are too big for production environment. I suggest we keep this code in a separate branch instead of merging into master. That way we can use this branch locally to showcase alongside our master branch which will be hosted online.

What is the file size requirement for production? 

This is an 'MVP' project. I would classify this un-optimized, but fully-functional 3d component to be of standard to a minimum viable product.

There are optimization libraries and manual compression of models. Optimization can be a separate issue.

File: client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs (hunk 3 above)
Comment (AGI-CEO) [reply client/vite.config.js.timestamp-1701470181193-448a84840ff72.mjs:1]: not required. will remove

**Comment by gnprwx** [comment]
They are fully-functional yes, but in a production environment they would need to be optimized first, especially if it was going to be hit with a heavy-volume of users. Big files like that are usually used in context of gif/video sites and are optimized to handle that. 

However in the context of this project, we can give it a shot on the azure deployment branch. I don't think it'll work well on render's free tier. But we could try that too if you want.

Once you're done with the changes, I'll test run it on the 3d branch, and work on the placement of the objects. If all is well, I'll try it on master so we can see what it looks like on render. I'm worried we'd destroy our bandwidth on render's free tier, but maybe that won't matter? We can only experiment.

**Pull request merged**